Source/Scene/Scene.js

```diff
@@ -2848,44 +3007,88 @@ define([
+            functionToExecute(scene, time);
+        } catch (error) {
+            scene._renderError.raiseEvent(scene, error);
+
+            if (scene.rethrowRenderErrors) {
+                throw error;
             }
+        }
+    }
 
-            scene._performanceDisplay.update();
-        } else if (defined(scene._performanceDisplay)) {
-            scene._performanceDisplay = scene._performanceDisplay && scene._performanceDisplay.destroy();
-            scene._performanceContainer.parentNode.removeChild(scene._performanceContainer);
+    function updateAndRenderScene(scene, time, forceRender) {
+        scene._jobScheduler.resetBudgets();
+
+        // Update
+        scene._preUpdate.raiseEvent(scene, time);
+        tryAndCatchError(scene, time, update);
+        scene._postUpdate.raiseEvent(scene, time);
+
+        var cameraChanged = checkForCameraUpdates(scene);
+        var shouldRender = !scene.requestRenderMode || forceRender || scene._renderRequested || cameraChanged || (scene.mode === SceneMode.MORPHING);
+        if (!shouldRender && defined(scene.maximumRenderTimeChange) && defined(scene._lastRenderTime)) {
+            var difference = Math.abs(JulianDate.secondsDifference(scene._lastRenderTime, time));
+            shouldRender = shouldRender || difference >= scene.maximumRenderTimeChange;
         }
 
-        context.endFrame();
-        RequestScheduler.update();
-        callAfterRenderFunctions(frameState);
+        if (shouldRender) {
+            scene._lastRenderTime = JulianDate.clone(time, scene._lastRenderTime);
+            scene._renderRequested = false;
+
+            // Render
+            scene._preRender.raiseEvent(scene, time);
+            tryAndCatchError(scene, time, render);
+            scene._postRender.raiseEvent(scene, time);
 
-        scene._postRender.raiseEvent(scene, time);
+            RequestScheduler.update();
+        }
+
+        updateDebugShowFramesPerSecond(scene, shouldRender);
+        callAfterRenderFunctions(scene);
     }
 
     /**
+     * Updates and renders the scene.
+     * @param {JulianDate} [time] The simulation time to render.
+     *
      * @private
      */
     Scene.prototype.render = function(time) {
-        try {
-            render(this, time);
-        } catch (error) {
-            this._renderError.raiseEvent(this, error);
+        if (!defined(time)) {
+            time = JulianDate.now();
+        }
 
-            if (this.rethrowRenderErrors) {
-                throw error;
-            }
+        updateAndRenderScene(this, time, false);
+    };
+
+    /**
+     * Updates and renders the scene. Always forces a new render frame regardless of if a render was requested.
+     * @param {JulianDate} [time] The simulation time to render.
+     *
+     * @private
+     */
+    Scene.prototype.forceRender = function(time) {
+        if (!defined(time)) {
+            time = JulianDate.now();
         }
+
+        updateAndRenderScene(this, time, true);
```

Comment:
Instead of introducing another param here, why not just `this._requestRendered = true`?

Reply:
You probably don't need the private `updateAndRenderScene` function either.  You could just call `this.render()` from this function after setting `this._requestRendered = true`.  Unless I'm missing something.